python/pyspark/sql/connect/client/core.py

```diff
@@ -666,7 +666,7 @@ def __init__(
         elif user_id is not None:
             self._user_id = user_id
         else:
-            self._user_id = os.getenv("USER", None)
+            self._user_id = os.getenv("SPARK_USER", os.getenv("USER", None))
```

Comment:
Hmm, why here is env `USER` not `user.name`?

Reply:
Btw, I think this PR fixes `SparkConnectClient`, and the description also shows `pyspark` respects default user when `--user_id` is not given. Why this change to `core.py` is needed?

Reply:
Thank you @dongjoon-hyun